bit_vector/bit_vector.py

```diff
@@ -400,17 +415,17 @@ def as_bool_list(self):
     def repeat(self, other):
         return BitVector( other.as_uint() * self.bits() )
 
-    @binary
+    @binary_no_cast
     def sext(self, other):
         return self.concat(BitVector(other.as_uint() * [self[-1]]), self)
 
-    @binary
+    @binary_no_cast
     def ext(self, other):
         return self.zext(other)
 
-    @binary
+    @binary_no_cast
```

Comment:
From what I understand, `@binary` just casts the type used as input (e.g. `BitVector` vs `SIntVector`). This shouldn't affect the `num_bits`, so do we need this change? The downside to this is that if we `zext` a `SIntVector`, this returns a `BitVector` instead of an `SIntVector`. This is okay since we can ask the user to make an explicit cast after using this function, but I'm wondering if it's necessary.

Reply:
I added a test `test_adc.py` and it still passes with the following changes
```
diff --git a/bit_vector/bit_vector.py b/bit_vector/bit_vector.py
index b65d02e..99f8c56 100644
--- a/bit_vector/bit_vector.py
+++ b/bit_vector/bit_vector.py
@@ -415,15 +415,15 @@ class BitVector:
     def repeat(self, other):
         return BitVector( other.as_uint() * self.bits() )
 
-    @binary_no_cast
+    @binary
     def sext(self, other):
         return self.concat(BitVector(other.as_uint() * [self[-1]]), self)
 
-    @binary_no_cast
+    @binary
     def ext(self, other):
         return self.zext(other)
 
-    @binary_no_cast
+    @binary
     def zext(self, other):
         return BitVector.concat(BitVector(other.as_uint() * [0]), self)
 
```

Can you provide an example where the extension doesn't work as expected with the original casting logic?

Reply:
Ok, let's go with this version. I was having trouble calling adc with two 16-bit numbers and a 1-bit number. I will retry.